OpenSim/Simulation/Model/KneeLigament.h

```diff
@@ -0,0 +1,237 @@
+ * The OpenSim API is a toolkit for musculoskeletal modeling and simulation.  *
+ * See http://opensim.stanford.edu and the NOTICE file for more information.  *
+ * OpenSim is developed at Stanford University and supported by the US        *
+ * National Institutes of Health (U54 GM072970, R24 HD065690) and by DARPA    *
+ * through the Warrior Web program.                                           *
+ *                                                                            *
+ * Copyright (c) 2005-2013 Stanford University and the Authors                *
+ * Author(s): Peter Loan, Dimitar Stanev                                      *
+ *                                                                            *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may    *
+ * not use this file except in compliance with the License. You may obtain a  *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.         *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing, software        *
+ * distributed under the License is distributed on an "AS IS" BASIS,          *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ * -------------------------------------------------------------------------- */
+
+
+//=============================================================================
+// INCLUDES
+//=============================================================================
+#include <OpenSim/Common/ScaleSet.h>
+#include "Model.h"
+#include "Force.h"
+
+#ifdef SWIG
+    #ifdef OSIMACTUATORS_API
+        #undef OSIMACTUATORS_API
+        #define OSIMACTUATORS_API
+    #endif
+#endif
+
+namespace OpenSim {
+
+class GeometryPath;
+
+//=============================================================================
+//=============================================================================
+/**
+A class implementing a knee ligament. The ligament is described by three
+parameters, rest length \f$ l_0 \f$, the stiffness \f$ k \f$ and toe region
+\f$ e_l \f$.
+
+The strain is given by:
+
+\f[
+e = (l -l_0) / l_0
+\f]
+
+where \f$ l_0 \f$ is the resting length of the ligament
+
+The force strain function is given by:
+
+\f[
+f(e) = 0, e \leq 0 
+\f]
+
+\f[
+f(e) = \frac{1}{4} \cdot \frac{e^2}{e_l}, e \leq 2 \cdot e_l
+\f]
+
+\f[
+f(e) = k \cdot (e - e_l), e > 2 \cdot e_l
+\f]
+
+where, \f$ k \f$ is the stiffness of the ligament and \f$ e_l \f$ is the limit 
+at which the ligament moves from the non-linear region to linear region.
+
+Sometime we define the reference length \f$ l_r \f$ and the reference strain
+\f$ e_r \f$, so the rest length is given by:
+
+\f[
+l_0 = l_r / (e_r + 1)
+\f]
+
```

Comment:
Given the analytical function you could easily compute the potential energy stored in the Ligament. This would allow for accurate energy accounting for verify the model and implementation. You can do this by overriding `double Force::computePotentialEnergy(const SimTK::State& state) const`